uber/models/attendee.py

```diff
@@ -307,14 +382,23 @@ def _use_promo_code(self):
 
     def unset_volunteering(self):
         self.staffing = False
-        self.trusted_depts = self.requested_depts = self.assigned_depts = ''
+        self.requested_depts = []
+        self.assigned_depts = []
         self.ribbon = remove_opt(self.ribbon_ints, c.VOLUNTEER_RIBBON)
         if self.badge_type == c.STAFF_BADGE:
             self.badge_type = c.ATTENDEE_BADGE
             self.badge_num = None
         del self.shifts[:]
 
     @property
+    def ribbon_labels(self):
```

Comment:
Wait, does this... add the Dept Head ribbon to anyone who's marked as a department head in a department, regardless of whether they have the Dept Head ribbon? Wouldn't it be a little more straightforward to just add a presave adjustment, if we want this behavior?

Reply:
Ok, so here's the deal with ribbons. The way we manage ribbons is not great. All of our currently defined ribbons would be better implemented as calculated properties. Lets take a look at our ribbons:

* "RockStar" - Is the attendee in a band, or the guest of a band?
* "Between 13 and 18 (unused)" - Is the attendee a teenager?
* "Under 13" - Is the attendee under the age of 13
* "Volunteer" - Is the attendee a volunteer?
* "Department Head" - Is the attendee a department head?
* "Shopkeep" - Is the attendee a dealer?
* "Panelist" - Is the attendee conducting a panel?

We have all of the data in our database to calculate the answers to those questions. Nobody should ever need to check a checkbox to make sure somebody is getting the right ribbons. Furthermore, nobody should ever be able to _uncheck_ a checkbox and accidentally remove a ribbon.

If you must mutate two variables in lock-step in order to maintain data integrity, that's a good sign that one of those variables should be derived from the other. Eventually, the department head ribbon checkbox should be removed entirely. That'll have to happen in a separate pull request in our production config.

Ultimately we should remove _ALL_ of those ribbon checkboxes.

Reply:
If we want to be able to grant someone a ribbon who wouldn't otherwise get that ribbon (like granting someone a dept head ribbon who isn't actually a dept head) then we can definitely leave the checkboxes in place!

I still prefer calculating the dept head ribbon as implemented, because it separates the reason _why_ someone is getting the ribbon. Consider the following example, where someone is *NOT* a dept head, but needs a dept head ribbon as you described:

1. The attendee is granted a dept head ribbon by manually checking the box
2. Later, they are made an actual dept head in the system (possibly by accident)
3. Even later, the attendee is removed as a dept head in the system (again, possibly by accident)
4. **Result**: the attendee's dept head ribbon is removed by the `@presave_adjustment` even though we still want that attendee to have a dept head ribbon

We've overloaded the `ribbon` field to track two separate things: 
1. Ribbons that people deserve based on their status in the system
2. Ribbons that people have been manually granted

Overloading a single variable to track multiple states is a thing we do a lot in ubersystem, and it's something I'd like to get away from.

Reply:
Totally agreed on getting away from this overload!

I was thinking of only having a presave adjustment that adds the ribbon, not removes it... but this is totally fine. :)